installed-tests/js/testGObjectInterface.js

```diff
@@ -0,0 +1,323 @@
+// -*- mode: js; indent-tabs-mode: nil -*-
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const JSUnit = imports.jsUnit;
+const Lang = imports.lang;
+const Mainloop = imports.mainloop;
+
+const MyInterface = new Lang.Interface({
+    Name: 'MyInterface',
+});
+
+const MyInterfaceGObject = new Lang.Class({
+    Name: 'MyInterfaceGObject',
+    Extends: GObject.Object,
+    Implements: [ MyInterface ],
+
+    _init: function (props={}) {
+        this.parent(props);
+    }
+});
+
+const MyGObjectInterface = new Lang.Interface({
+    Name: 'MyGObjectInterface',
+    GTypeName: 'ArbitraryGTypeName',
+    Requires: [ GObject.Object ],
+    Properties: {
+        'interface-prop': GObject.ParamSpec.string('interface-prop',
+            'Interface property', 'Must be overridden in implementation',
+            GObject.ParamFlags.READABLE,
+            'foobar')
+    },
+    Signals: {
+        'interface-signal': {}
+    },
+
+    requiredG: Lang.Interface.UNIMPLEMENTED,
+    optionalG: function () {
```

Comment:
Implementing `GInterface` vfuncs is optional. Just how optional depends on the interface writer. For example, if you just leave a null entry in the vtable then the interface might log a critical saying "You must implement this function". That's the equivalent of `Lang.Interface.UNIMPLEMENTED` here.

On the other hand, many interfaces provide default implementations of their vfuncs. For example, `GFile` does not require you to implement all of its async vfuncs, because the default implementation is just to run the sync vfunc in a thread.